src/Mono.Android/Android.Runtime/JNIEnv.cs

```diff
@@ -322,6 +323,21 @@ public static IntPtr FindClass (System.Type type)
 			}
 		}
 
+		internal static JniObjectReferenceOptions ToJniObjectReferenceOptions (JniHandleOwnership transfer)
+		{
+			switch (transfer) {
```

Comment:
`JniHandleOwnership` & `JniObjectReferenceOptions` are both `[Flags]` enums.  Instead of `case` statements, this should instead check for flags:

```csharp
JniObjectReferenceOptions options = default;
if (transfer.HasFlag (JniHandleOwnership.TransferLocalRef) ||
        transfer.HasFlag (JniHandleOwnership.TransferGlobalRef))
    options |= JniObjectReferenceOptions.CopyAndDispose;
if (transfer.HasFlag (JniHandleOwnership.DoNotRegister))
    options |= JniObjectReferenceOptions.CopyAndDoNotRegister;
if (transfer == JniHandleOwnership.DoNotTransfer) // 0; can't use with .HasFlag()
    options |= JniObjectReferenceOptions.Copy;

return options;
```